hugegraph-core/src/main/java/com/baidu/hugegraph/backend/page/PageInfo.java

```diff
@@ -101,10 +101,26 @@ public static PageInfo fromBytes(byte[] bytes) {
         }
     }
 
-    public static String page(Iterator<?> iterator) {
+    public static PageState pageState(Iterator<?> iterator) {
         E.checkState(iterator instanceof Metadatable,
                      "Invalid paging iterator: %s", iterator.getClass());
         Object page = ((Metadatable) iterator).metadata(PAGE);
-        return (String) page;
+        E.checkState(page instanceof PageState,
+                     "Invalid PageState '%s'", page);
+        return (PageState) page;
+    }
+
+    /**
+     * Only used for vertex/edge scan
+     * @param iterator iterator of scan query
+     * @return PageInfo string
+     */
+    public static String pageInfo(Iterator<?> iterator) {
+        E.checkState(iterator instanceof Metadatable,
+                     "Invalid paging iterator: %s", iterator.getClass());
+        Object page = ((Metadatable) iterator).metadata(PAGE);
+        E.checkState(page == null || page instanceof PageInfo,
```

Comment:
PageInfo won't be null

Reply:
null is returned if offset > total for PageInfo